sdk/core/azure-core/tests/test_retry_policy.py

```diff
@@ -159,3 +160,31 @@ def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineRe
         pipeline = Pipeline(MockTransport(), [http_retry])
         pipeline.run(http_request)
     os.unlink(f.name)
+
+def test_retry_timeout():
+    class MockTransport(HttpTransport):
+        def __init__(self):
+            self.count = 0
+
+        def __exit__(self, exc_type, exc_val, exc_tb):
+            pass
+        def close(self):
+            pass
+        def open(self):
+            pass
+
+        def send(self, request, **kwargs):  # type: (PipelineRequest, Any) -> PipelineResponse
+            self.count += 1
+            if self.count > 2:
+                assert self.count <= 2
+            time.sleep(5)
+            raise ServiceResponseError('timeout')
+
+    http_request = HttpRequest('GET', 'http://127.0.0.1/')
+    headers = {'Content-Type': "multipart/form-data"}
+    http_request.headers = headers
+    http_retry = RetryPolicy(retry_total=10, timeout=10)
```

Comment:
This test is designed to take 10seconds, I usually try to avoid such big values since eventually if we have too much test like that CI starts to become slow.

I understand time is an important factor here, but I would encourage to make smaller values. If you're total timeout is 1sec, and you retry every 500ms, you still test your behavior but it's 10X faster.